docs/docs/user/references/components.md

```diff
@@ -0,0 +1,245 @@
+configuration. To learn more about their hierarchy, visit the
+[architecture](/docs/developer/architecture/component-inheritance.md) page.
+
+## KubernetesApp
+
+### Usage
+
+Can be used to deploy any app in Kubernetes using Helm, for example, a REST
+service that serves Kafka data.
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/kubernetes-app.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+Deploy using Helm.
+
+#### destroy
+
+Uninstall Helm release.
+
+#### reset
+
+Not implemented.
+
+#### clean
+
+Not implemented.
+
+## KafkaApp
+
+Sub class of [_KubernetesApp_](#kubernetesapp).
+
+### Usage
+
+- Deploys a [streams-bootstrap](https://github.com/bakdata/streams-bootstrap)
+component
+- Should not be used in `pipeline.yaml` as the component can be defined as either a
+[StreamsApp](#streamsapp) or a [Producer](#producer)
+- Often used in `defaults.yaml`
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/kafka-app.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+In addition to [KubernetesApp's](#kubernetesapp) `deploy`:
+
+- Create topics if provided (optional)
+- Submit schemas to the registry if provided (optional)
+
+#### destroy
+
+Refer to [KubernetesApp](#kubernetesapp).
+
+#### reset
+
+Not implemented.
+
+#### clean
+
+Not implemented.
+
+## StreamsApp
+
+Sub class of [_KafkaApp_](#kafkaapp).
+
+### Usage
+
+Configures a
+[streams-bootstrap app](https://github.com/bakdata/streams-bootstrap) which in
+turn configures a
+[Kafka Streams app](https://kafka.apache.org/documentation/streams/).
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/streams-app.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+Refer to [KafkaApp](#kafkaapp).
+
+#### destroy
+
+Refer to [KafkaApp](#kafkaapp).
+
+#### reset
+
+Runs a reset job in the cluster that is deleted by default after a succesful
+completion.
+
+- Reset the consumer group offsets
+- Reset Kafka Streams state
+
+#### clean
+
+- Reset Kafka Streams state
+- Delete the app's output topics
+- Delete all associated schemas in the Schema Registry
+- Delete the consumer group
+
+## Producer
+
+Sub class of [_KafkaApp_](#kafkaapp).
+
+### Usage
+
+Publishes (writes) events to a Kafka cluster.
+
+### Configuration
+
+??? "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/producer.yaml
+      --8<--
+    ```
+
+### Operations
+
+#### deploy
+
+Refer to [KafkaApp](#kafkaapp).
+
+#### destroy
+
+Refer to [KafkaApp](#kafkaapp).
+
+#### reset
+
+Not implemented, producers are stateless.
+
+#### clean
+
+- Delete the output topics of the Kafka producer
+- Delete all associated schemas in the Schema Registry
+
+## KafkaSinkConnector
+
+Sub class of [KafkaConnector](/docs/user/references/config.md)
```

Comment:
Does not exist (yet?)

Reply:
No. Supposed to be the future yamls page. The idea is that you would use `KafkaKonnector` only in `defaults.yaml` with which we are not concerned in "Components"/

Reply:
Then remove the broken link for now.

Reply:
and mark as todo somehow please :)

Reply:
done